spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/EntityPropertyValueProvider.java

```diff
@@ -33,8 +36,19 @@
 public class EntityPropertyValueProvider implements PropertyValueProvider<DatastorePersistentProperty> {
 	private Entity entity;
 
-	EntityPropertyValueProvider(Entity entity) {
+	private ConversionService conversionService;
```

Comment:
This `conversionService` may be different from the one in `DefaultDatastoreEntityConverter` which seems to be a potential problem. I think that conversion logic for reading and for writing should be in the same class, not spread out between `EntityPropertyValueProvider` and `DefaultDatastoreEntityConverter`.